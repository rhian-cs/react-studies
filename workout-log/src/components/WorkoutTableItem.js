import React from 'react';

const WorkoutTableItem = ({ uuid, hours, type, date, onDestroy }) => (
  <tr>
    <td>{hours}h</td>
    <td>{type}</td>
    <td>{date}</td>
    <td>
      <button onClick={() => onDestroy(uuid)}>Delete</button>
    </td>
  </tr>
);

export default WorkoutTableItem;
