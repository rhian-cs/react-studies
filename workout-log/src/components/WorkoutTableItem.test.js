/**
 * @jest-environment jsdom
 */

import React from 'react';
import '@testing-library/jest-dom';

import WorkoutTableItem from './WorkoutTableItem';

import { render } from '@testing-library/react';

describe('WorkoutTableItem', () => {
  const props = {
    uuid: '00001234',
    hours: '2',
    type: 'bicycle',
    date: '2019-02-03',
    onDestroy: () => {},
  };

  it('matches snapshot', () => {
    const component = render(
      <table>
        <tbody>
          <WorkoutTableItem {...props} />
        </tbody>
      </table>,
    );

    expect(component.container).toMatchSnapshot();
  });

  it.todo('triggers onDestroy action when clicking on button');
});
