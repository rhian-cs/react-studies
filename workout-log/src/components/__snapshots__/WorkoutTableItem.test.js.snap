// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkoutTableItem matches snapshot 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td>
          2
          h
        </td>
        <td>
          Bicycle
        </td>
        <td>
          2019-02-03
        </td>
        <td>
          <button
            class="delete-button"
          >
            â€“
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
